src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -960,7 +961,7 @@ def add_relates_to(self, relates_to: str) -> AudioSegmentResource:
         self.relates_to.append(relates_to)
         return self
 
-    def add_relates_to_multiple(self, relates_to: list[str]) -> AudioSegmentResource:
+    def add_relates_to_multiple(self, relates_to: Sequence[str]) -> AudioSegmentResource:
```

Comment:
Looks good, you can now delete the line 1106 & 5.

Reply:
I'm not sure. AFAIU, `_transform_unexpected_input()` is now completely obsolete everywhere, if we assume that our users exclusively use our public API. I mean, if users only use the constructor and the public methods, there is no code path where there could be a wrong data type in one of these fields.

But as soon as someone manually modifies the class fields, there might be bad surprises. So we could also leave `_transform_unexpected_input()` in place, as an additional layer of security. WDYT?

Reply:
Sounds good, it was meant to make it more stable and if you hand in a string iteration will not raise an error...